### ISSUE TITLE
[TS] Optional parameters for Phaser.Group.addAll()

### ISSUE THREAD

**Issue by alexus85** [photonstorm/phaser-ce#45]
This Issue is about :  **an error in the TypeScript defs**

( _Phaser  v. 2.6.2_)

Just converted my project into typescript and I hit couple of TS errors while trying to update some values for all my `Phaser.Group` elements  using `.addAll() ` method.

**Below example should throw an error**

`this.asteroidsWave.addAll('angle', this.game.rnd.integerInRange(1,4));`

**Error:**
error TS2346: Supplied parameters do not match any signature of call target.

**working example:**

`this.asteroidsWave.addAll('angle', this.game.rnd.integerInRange(1,4), false, false);`

After reading the TS definitions in phaser.d.ts , it turned out that the last two parameters aren't optional , which I think they should be , cause when they're not defined they're implicitly changed to `false`. 

**phaser.js**
``` 
33358: Phaser.Group.prototype.setAll = function (key, value, checkAlive, checkVisible, operation, force) {
    if (checkAlive === undefined) { checkAlive = false; }
    if (checkVisible === undefined) { checkVisible = false; }
    if (force === undefined) { force = false; }
```

**phaser.d.ts**
`1768:   addAll(property: string, amount: number, checkAlive: boolean, checkVisible: boolean): void;`

Either setting the default values for last two parameters or making them optional should fix the error.

` addAll(property: string, amount: number, checkAlive?: boolean, checkVisible?: boolean): void;
`
or 

` addAll(property: string, amount: number, checkAlive: boolean = false, checkVisible: boolean=false): void;
`

**Comment by Dreaded-Gnu** [comment]
Comparing the docs, this doesn't seem to be an error. Documentation says that setAll has 4 optional parameters, which is also at typescript definitions.

The addAll methods doesn't have any optional parameter within the docs. When making the parameters optional at javascript also fallbacks have to be added within the function addAll. That means, that also at subAll optional parameters have to be set with fallbacks in javscript to be consistent.

Or am I wrong at this point?

**Comment by samme** [comment]
@alexus85 Can you add a pull request here?

**Comment by alexus85** [comment]
#113 , #115

**Comment by samme** [comment]
Fixed in #112, #115, #123
